nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -151,11 +151,13 @@ in
       };
 
       daemonIONiceLevel = mkOption {
-        type = types.int;
+        type = types.either types.int (types.enum ["idle"]);
```

Comment:
There's a pretty flexible system for dealing with that, see https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/rename.nix, it supports e.g., mapping old option declarations to new ones, establishing aliases, &c